documentation/rules/NS5002.md

```diff
@@ -0,0 +1,54 @@
+﻿# NS5002
+
+<table>
+<tr>
+  <td>CheckId</td>
+  <td>NS5002</td>
+</tr>
+<tr>
+  <td>Category</td>
+  <td>Usage</td>
+</tr>
+</table>
+
+## Cause
+
+Usage of async callback in `Received.InOrder` method.
+
+## Rule description
+
+A violation of this rule occurs when async callback is used in `Received.InOrder` method.
```

Comment:
I think we should add a description of why this is a problem. Maybe just link to the issue?

> A violation of this rule occurs when async callback is used in `Received.InOrder` method.  `Received.InOrder` is used to specify calls which should be received by one or more substitutes. Running/awaiting these calls is not required for this, and can cause some problems as described in issue [#604](https://github.com/nsubstitute/NSubstitute/issues/604).


```markdown
A violation of this rule occurs when async callback is used in `Received.InOrder` method.  `Received.InOrder` is used to specify calls which should be received by one or more substitutes. Running/awaiting these calls is not required for this, and can cause some problems as described in issue [#604](https://github.com/nsubstitute/NSubstitute/issues/604).
```

Reply:
Fixed